sphinx/writers/latex.py

```diff
@@ -2163,26 +2147,36 @@ def depart_title_reference(self, node):
         # type: (nodes.Node) -> None
         self.body.append('}')
 
+    def visit_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        longest_label = max((subnode[0].astext() for subnode in node), key=len)
+        self.body.append(u'\n\\begin{sphinxthebibliography}{%s}\n' % longest_label)
+
+    def depart_thebibliography(self, node):
+        # type: (nodes.Node) -> None
+        self.body.append(u'\\end{sphinxthebibliography}\n')
+
     def visit_citation(self, node):
         # type: (nodes.Node) -> None
-        # TODO maybe use cite bibitems
-        # bibitem: [citelabel, citetext, docname, citeid]
-        self.bibitems.append(['', '', '', ''])
-        self.context.append(len(self.body))
+        label = node[0].astext()
+        self.body.append(u'\\bibitem[%s]{%s:%s}{' %
+                         (self.encode(label), node['docname'], node['ids'][0]))
```

Comment:
with single doc source called `index.rst` this creates link ids of the type `index:foo`. I don't see potential for trouble here, because `\cite` is not `\hyperref`.

I checked the blank from end of line when citation itself gets inserted does not show. The opening brace `{` is not required by LaTeX. It can be suppressed (then of course also closing brace from depart_citation.

Reply:
Okay, I removed braces.